src/main/java/io/confluent/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -229,6 +293,26 @@ private String getInsertSql() {
     }
   }
 
+  private String getDeleteSql() {
+    String sql = null;
+    if (config.deleteEnabled) {
+      switch (config.pkMode) {
+        case RECORD_KEY:
+          if (fieldsMetadata.keyFieldNames.isEmpty()) {
+            throw new ConnectException("Require primary keys to support delete");
+          }
+          sql = dbDialect.buildDeleteStatement(
```

Comment:
Should we catch `UnsupportedOperationException` and throw a `ConnectException` with a more useful message?

Reply:
done